### PR TITLE
Integrate About Automattic Library

### PR DESCRIPTION
**Pull request by ravishanker**
This PR integrates About Automattic library into WPAndroid and uses its implementation to replace current About Activity when feature is enabled.  Also includes setup for composite local build.

Fixes #15529

To test:
- Enable `UnifiedAboutFeatureConfig` from Me -> App Settings -> Debug Settings and restart
- Go to Me -> App Settings -> scroll down and click on WordPress for Android under About the app
- See the new About from the library displayed
- Click on Legal and more, it navigates to Legal and More screen

NOTE:  After much wrangling a number of build issues, lint checks and checkstylle issues have been fixed in this PR and also in a separate [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/15602) that is merged in.

## Regression Notes
1. Potential unintended areas of impact
Fixed all known issues

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Existing unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: local-builds.gradle-example
Comment (mokagio) [inline local-builds.gradle-example]: Nitpick: Given the name of the [repo](https://github.com/Automattic/about-automattic-android) for the library is `about-automattic-android`, I'd suggest using that here.

```suggestion
    //localAboutAutomatticPath = "../about-automattic-android"
```

File: settings.gradle
Comment (mokagio) [inline settings.gradle]:
```suggestion
                println "Substituting about-automattic with the local build"
```

File: settings.gradle
Comment (mokagio) [inline settings.gradle]: When I use a local build, this fails:

```
Substituting about-automatiic with the local build

FAILURE: Build failed with an exception.

* Where:
Settings file '/Users/gio/Developer/a8c/wpandroid/settings.gradle' line: 112

* What went wrong:
Project :AboutAutomattic not found.
```

This is the `local-builds.gradle` I'm using

```
ext {
    localAboutAutomatticPath = "../about-automattic-android"
    //localFluxCPath = "../WordPress-FluxC-Android"
    //localWPUtilsPath = "../WordPress-Utils-Android"
    //localGutenbergMobilePath = "../gutenberg-mobile"
    //localLoginFlowPath = "../WordPress-Login-Flow-Android"
    //localStoriesAndroidPath = "../stories-android"
    //localAztecAndroidPath = "../AztecEditor-Android"
}
```

And I verified the path is correct 😄 : 

image

**Review by mokagio — COMMENTED** [review]
Not sure what the source of the lint failures might be and how it relates to these changes (if at all), but I'm getting it locally, too.

File: settings.gradle
Comment (ravishanker) [reply settings.gradle]: I found the issue with the above.  Looks like it was renamed to `library` in `automattic-android-about`  repo hence it is not finding it.  I changed to :library at line 112 in settings.gradle that resolves it. 

However, that's not the reason for lint failure though

Comment (ravishanker) [reply settings.gradle]: The issue could be with this

`Could not load custom lint check jar file .gradle/caches/transforms-3/03a4dea5bd244b5ca160f0ea25e79cb3/transformed/jetified-animation-core-1.1.0-beta02/jars/lint.jar`

We've just started using Jepack Compose (a.k.a android SwiftUI).  Lint seems to have problem with it.  

I'm not sure why it is not a problem on the about-automattic-android library where it is used!  Also, on WPAndroid detekt and ktlint seems to pass fine!

Comment (renanferrari) [reply settings.gradle]: Looking at the lint results document, it looks like there are several failures going on. One of them produces the following error message:

```
  | [2021-11-19T05:06:42Z]   /workdir/WordPress/src/main/java/org/wordpress/android/push/NativeNotificationsUtils.java:70: Error: Missing permissions required by intent Intent.ACTION_CLOSE_SYSTEM_DIALOGS: android.permission.BROADCAST_CLOSE_SYSTEM_DIALOGS [MissingPermission]
  | [2021-11-19T05:06:42Z]           context.sendBroadcast(closeIntent);
  | [2021-11-19T05:06:42Z]                   ~~~~~~~~~~~~~~~~~~~~~~~~~
  | [2021-11-19T05:06:42Z]   /workdir/WordPress/src/main/java/org/wordpress/android/ui/notifications/ShareAndDismissNotificationReceiver.java:21: Error: Missing permissions required by intent Intent.ACTION_CLOSE_SYSTEM_DIALOGS: android.permission.BROADCAST_CLOSE_SYSTEM_DIALOGS [MissingPermission]
  | [2021-11-19T05:06:42Z]           context.sendBroadcast(new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS));
  | [2021-11-19T05:06:42Z]                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This seems to indicate that this particular failure occurs because we're missing the permission to close system dialogs, [which is a requirement for Android 12](https://developer.android.com/about/versions/12/behavior-changes-all#close-system-dialogs). According to that page, this change affects **all apps**: 

>The following behavior changes apply to _all apps_ when they run on Android 12, regardless of `targetSdkVersion`. You should test your app and then modify it as needed to support these properly, where applicable.

If I had to guess, I'd say these issues are now visible because we bumped `compileSdkVersion` to `31`. Maybe we should handle those changes on a separate PR before merging this.

Let me know what you think.

Comment (ravishanker) [reply settings.gradle]: I thought so, `compileSdkVersion` change could be the reason.  Then I tested it on develop branch without that change and it's happening on there too!

File: WordPress/build.gradle
Comment (oguzkocer) [inline WordPress/build.gradle]: We don't have a `com.automattic.about` group. `about-automattic-android` library is published under `com.automattic` group as described [in the setup instructions in the repository's README.](https://github.com/Automattic/about-automattic-android/blob/main/README.md#use-the-library-in-your-project) Since `com.automattic` group is already included, no further change is necessary.

```suggestion
```

**Review by oguzkocer — COMMENTED** [review]
@ravishanker I had a quick look at this PR while reviewing #15602 and noticed a minor issue and left a suggestion for it.

File: config/checkstyle.xml
Comment (renanferrari) [inline config/checkstyle.xml:101]: I suppose we're changing the severity of this issue because we can't suppress it otherwise?

File: build.gradle
Comment (renanferrari) [inline build.gradle]: Should this point to the main branch before we merge this?

File: WordPress/src/main/java/org/wordpress/android/ui/about/AboutWordPressActivity.kt
Comment (renanferrari) [inline WordPress/src/main/java/org/wordpress/android/ui/about/AboutWordPressActivity.kt]: Notes for future PRs: 

- We should consider implementing this with a proper `ViewModel`.
- We should consider which values to provide when building the Jetpack app.

**Review by renanferrari — APPROVED** [review]
Thanks for taking care of this! I tested the app and everything is working as expected. 

I left a few comments and questions below, but none of them are blockers, so feel free to merge this after addressing them.

LGTM 👍

File: config/checkstyle.xml
Comment (ravishanker) [reply config/checkstyle.xml:101]: Correct

File: build.gradle
Comment (ravishanker) [reply build.gradle]: Yes

**Pull request merged**
